### PR TITLE
 Widget Grouping Context Menu

### LINKED ISSUE #5013: Context Menu for selected widgets
- [x] show selected widgets with context menu for actions
- [x] highlight selected widgets by showing a border around them
- [x] enable existing bulk operations implemented(Cut/Copy/Paste)

https://www.figma.com/file/okCgRwbn6qazHVLmh7MGnU/Widget-Grouping?node-id=665%3A574

### PR DESCRIPTION
**Pull request by jsartisan**
This PR adds the context menu in widget grouping. Now a context menu will appear when multiple widgets are selected to perform actions like ( cut /copy/ delete )

Fixes #5013 #5679 #4993 #5422

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Updated cypress tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/widget-grouping-3-context-menu 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.02 **(0.11)** | 36 **(0.15)** | 32.73 **(0.18)** | 54.6 **(0.1)**
 :green_circle: | app/client/src/components/designSystems/appsmith/ContainerComponent.tsx | 93.33 **(0)** | 59.09 **(4.09)** | 100 **(0)** | 93.33 **(0)**
 :green_circle: | app/client/src/icons/ControlIcons.tsx | 57.89 **(1.71)** | 100 **(0)** | 13.04 **(3.74)** | 57.89 **(1.71)**
 :sparkles: :new: | **app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx** | **86.67** | **88.89** | **79.17** | **85.54**
 :green_circle: | app/client/src/sagas/WidgetOperationSagas.tsx | 44.4 **(0.32)** | 30.72 **(0.44)** | 50.59 **(0)** | 44.65 **(0.19)**
 :green_circle: | app/client/src/sagas/selectors.tsx | 72.73 **(0.69)** | 54.17 **(-0.38)** | 50 **(1.85)** | 66.67 **(0.47)**
 :red_circle: | app/client/src/utils/hooks/useClickOpenPropPane.tsx | 81.13 **(0)** | 52.38 **(-0.25)** | 100 **(0)** | 80 **(0)**
 :green_circle: | app/client/src/widgets/ContainerWidget.tsx | 69.81 **(0.58)** | 50 **(0)** | 70 **(0)** | 72 **(0.57)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+  position: relative;
+  height: 100%;
+  width: 100%;
+`;
+
+const StyledActions = styled.div`
+  margin-left: calc(100% + 4px);
+  padding: 5px 0;
+  width: max-content;
+  z-index: 4;
```
Comment (marks0351) [inline app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx:36]: Can we have this added to layers constant?

File: app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+          {/* copy widgets */}
+          <Tooltip
+            boundary="viewport"
+            content={copyHelpText}
+            maxWidth="400px"
+            modifiers={PopoverModifiers}
+            position={Position.RIGHT}
+          >
+            <StyledAction
+              onClick={stopEventPropagation}
```
Comment (marks0351) [inline app/client/src/pages/Editor/WidgetsMultiSelectBox.tsx:289]: is this needed for the other actions as well?

**Review by marks0351 — CHANGES_REQUESTED** [review]

**Review by marks0351 — DISMISSED** [review]

**Comment by vivekverma2312** [comment]
@jsartisan 
1. Cut widgets in container and then paste it — it is getting pasted outside the container
https://www.loom.com/share/134d01dcc29b43e9836618a4a70fc901

2. Context menu is not visible if widgets are selected inside container/Form which at the corner and scroll is enabled
Screenshot 2021-06-30 at 4 16 47 PM

**Comment by vivekverma2312** [comment]
@jsartisan Deploy preview has been tested and working fine.

**Pull request merged**